browser/brave_shields/android/cookie_list_opt_in_service_factory_android.cc

```diff
@@ -0,0 +1,25 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+#include "base/android/jni_android.h"
+#include "brave/browser/brave_shields/cookie_list_opt_in_service_factory.h"
+#include "brave/build/android/jni_headers/CookieListOptInServiceFactory_jni.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "mojo/public/cpp/bindings/pending_receiver.h"
+
+namespace chrome {
+namespace android {
+static jint
+JNI_CookieListOptInServiceFactory_GetInterfaceToCookieListOptInService(
+    JNIEnv* env) {
+  auto* profile = ProfileManager::GetActiveUserProfile();
```

Comment:
we should pass a current profile from Android, check this
https://github.com/brave/brave-core/blob/master/browser/brave_wallet/android/keyring_service_factory_android.cc#L18
and this
https://github.com/brave/brave-core/blob/master/android/java/org/chromium/chrome/browser/crypto_wallet/KeyringServiceFactory.java#L36